### ISSUE TITLE
Thermal Expansion - Won't Pathfind Because of NPE

### ISSUE THREAD

**Issue by theoparis** [cabaletta/baritone#1294]
## Some information
Operating system: Mac OSX High Serria
Java version: 8
Minecraft version: 1.12.2
Baritone version: v1.2.11 1.12.2
Forge mods (if used): a lot

## Exception, error or logs
Please find your `latest.log` or `debug.log` in this folder and attach it to the issue
[latest.log](https://github.com/cabaletta/baritone/files/4196017/latest.log)

Linux: `~/.minecraft/logs/`

Windows: `%appdata%/.minecraft/logs/`

Mac: `/Library/Application\ Support/minecraft/logs/`

It spams a nullpointerexception error in the chat AND the console when i try to pathfind using the path command. In the console it says 

> [17:14:34] [pool-7-thread-4/INFO] [STDERR]: [baritone.hh:a:-1]: 	at cofh.thermalexpansion.block.BlockTEBase.func_176195_g(BlockTEBase.java:114)

I'm assuming it has to do with thermal expansion, and I haven't seen an issue on this yet.

**Comment by creepinson** [comment]
Mods:
![explorer_l6XRGEZ6AD](https://user-images.githubusercontent.com/11761863/74500392-98486480-4e9b-11ea-857a-1a45cf74664d.png)
![explorer_dedCLTgm6d](https://user-images.githubusercontent.com/11761863/74500396-98e0fb00-4e9b-11ea-86cd-f9436b7d0601.png)

**Comment by S-B99** [comment]
https://github.com/ImpactDevelopment/ImpactIssues/wiki/Usage-FAQ#impact-crashesdoes-something-weird-when-i-run-it-with-insert-amount-here-other-forge-mods

I bruh

definitely don't expect it to be fixed with a mod list like that. the link still applies

**Comment by S-B99** [comment]
tbh sometimes I wonder how people even get that many mods working together in the first place
you'd think at least one of them would write shitty code and make an incompatibility

**Comment by creepinson** [comment]
Well the error in the log shows to be a problem specifically with Thermal Expansion, and it's tile entity classes.

**Comment by S-B99** [comment]
idk man
probably not gonna get fixed if it's another forge mod causing it
just putting it out there

**Comment by 0-x-2-2** [comment]
@S-B99 >posting impact faq in baritone issue tracker
🤡

**Comment by S-B99** [comment]
still applies tho

**Comment by 0-x-2-2** [comment]
false

**Comment by 0-x-2-2** [comment]
> tbh sometimes I wonder how people even get that many mods working together in the first place
> you'd think at least one of them would write shitty code and make an incompatibility

its called not using coremods.

**Comment by S-B99** [comment]
then fix it lol

**Comment by 0-x-2-2** [comment]
> then fix it lol

what do you think the issue tracker is for.

**Comment by S-B99** [comment]
😳😳

**Comment by 0-x-2-2** [comment]
https://github.com/cabaletta/baritone/blob/5e9b43959147b14ec2d5775f74df73c3e4d6e6ed/src/main/java/baritone/utils/ToolSet.java#L157

TLDR baritone strikes again with giving invalid data

**Comment by creepinson** [comment]
Ok so I was investigating this, and the world parameter seems to be easy to handle, but the position is a problem because the toolset method that is called calculateSpeedVsBlock does not contain a position parameter because the usages of this method do not have a position either, and just a state. I'm not sure what to do either here.
Actually, if I pass in the primary baritone's player world, and then can I pass in a blockpos of 0,0,0?

**Comment by creepinson** [comment]
So, I don't know if my code below is good, but it seems to fix the spam of NPE errors. The stairs issue in the other issue is still happening though.

`java
float hardness = state.getBlockHardness(BaritoneAPI.getProvider().getPrimaryBaritone().getPlayerContext().world(), new BlockPos(0, 0, 0));

`

**Comment by creepinson** [comment]
Do you guys think my way of getting the world object will work in all cases, or should I find a better way?

**Comment by ZeroMemes** [comment]
Duplicate of #3945
